packages/dev/core/src/FlowGraph/flowGraphPath.ts

```diff
@@ -0,0 +1,82 @@
+import type { FlowGraphContext } from "./flowGraphContext";
+
+const SEPARATORS = /\/|\./;
+/*
+ * This class represents a path of type /x/{y}/z/.../w that is evaluated
+ * on a target object. The string between curly braces ({y} in the example)
+ * is a special template string that is replaced during runtime.
+ */
+export class FlowGraphPath {
+    private path: string;
+    private templateSubstitutions: {
+        [key: string]: number;
+    } = {}; // this is a map of template strings to values that are substituted during runtime
+    public hasTemplateStrings: boolean = false;
+
+    constructor(path: string) {
+        this.path = path;
+        const templateStrings = this._getTemplateStringsInPath(path);
+        this.hasTemplateStrings = templateStrings.length > 0;
+        for (const templateString of templateStrings) {
+            // Part of the path enclosed in curly braces = template string, we'll set a default
+            // value on the template substitutions for now
+            this.templateSubstitutions[templateString] = -1;
+        }
+    }
+
+    getTemplateStrings(): string[] {
+        return Object.keys(this.templateSubstitutions);
+    }
+
+    private _getTemplateStringsInPath(path: string): string[] {
+        const splitPath = path.split(SEPARATORS).filter((part) => part !== "");
+        const templateStrings: string[] = [];
+        for (const partPath of splitPath) {
+            if (partPath.startsWith("{") && partPath.endsWith("}")) {
+                templateStrings.push(partPath.slice(1, partPath.length - 1));

```

Comment:
Is it not allowed to use the same template string name several times in a path?

If yes, you should not push the name if it's already in `templateStrings`.